docs/security/trust/content_trust.md

```diff
@@ -0,0 +1,262 @@
+
+# Content trust in Docker
+
+When transferring data among networked systems, *trust* is a central concern.
+When using Docker you push and pull images (data) to a registry. Currently,
+trust is supported for users of Docker's public Hub. It is not available for
+Docker Trusted Registry or for private registries. 
+
+## Understand trust in Docker 
+
+Content trust allows operations with a remote Docker registry to enforce
+client-side signing and verification of image tags. Currently, content trust is
+disabled by default. You must enabled it by setting the `DOCKER_CONTENT_TRUST`
+variable.
+
+Once trust is enabled, image creators can sign their images. Image consumers can
+ensure that the images they use are signed. Creators and consumers can be
+individuals alone or in organizations. Docker's trust feature supports users and
+automated processes such as builds. 
+
+### Image tags and trust
+
+An individual image record has the following identifier:
+
+```
+[REGISTRY_HOST[:REGISTRY_PORT]/]REPOSITORY[:TAG]
+```
+
+A particular image `REPOSITORY` can have multiple tags. For example, `latest` and
+ `3.1.2` are both tags on the `mongo` image. An image creator can build a image
+ and tag combination many times changing the image with each build. 
+
+Trust applies to the `TAG` portion of an image. Each image repository has a set
+of keys that image creators use to sign an image tag. Image creators have
+discretion on which tags they sign.
+
+An image repository can contain an image with one tag that is signed and another
+tag that is not. For example, consider [the Mongo image
+repository](https://registry.hub.docker.com/_/mongo/tags/manage/). The `latest`
+tag could be unsigned while the `3.1.6` tag could be signed. It is the
+responsibility of the image creator to determine if an image tag is signed or
+not. In this representation, some image tags are signed, others are not:
+
+![Signed tags](../images/tag_signing.png)
+
+Currently, it is also possible to sign some image tag pushes and not others. For
+example, a creator can push a tagged image `someimage:latest` and sign it.
+Later, a creator can push and not sign `someimage:latest`. This is useful if
+creators want to iterate over a image before signing it's tag officially.
+
+Image consumers can enable trust to ensure that images they use were signed. If
+a consumer enables trust, they can only pull or run trusted images. Enabling
+trust is like wearing a pair of rose-colored glasses. Consumers "see" only
+signed images tags and the less desirable, unsigned image tags are "invisible" to them.
+
+![Trust view](../images/trust_view.png)
+
+To the consumer who is not using trust, every image is visible and can be consumed.
+
+
+### Trust operations and keys
+
+When trust is enabled, `docker` CLI commands that operate on tagged images must
+either have content signatures or explicit content hashes. The commands that
+operate with trust are:
+
+* `push`
+* `build`
+* `create` 
+* `pull`
+* `run` 
+
```

Comment:
Maybe add an example of what we mean by explicit content hashes? 

`For example, with content trust enabled a `docker pull someimage:latest` will only succeed if someimage:latest has been signed. However, an operation with an explicit content hash such as `docker pull someimage@sha256:d149ab53f8718e987c3a3024bb8aa0e2caadf6c0328f1d9d850b2a2a67f2819a` will always work as long as the content hash exists`